naff/client/client.py

```diff
@@ -1752,90 +1752,114 @@ async def _dispatch_interaction(self, event: RawGatewayEvent) -> None:
         else:
             raise NotImplementedError(f"Unknown Interaction Received: {interaction_data['type']}")
 
-    @Listener.create("message_create", is_default_listener=True)
-    async def _dispatch_prefixed_commands(self, event: MessageCreate) -> None:
+    @Listener.create("raw_message_create", is_default_listener=True)
+    async def _dispatch_prefixed_commands(self, event: RawGatewayEvent) -> None:
         """Determine if a prefixed command is being triggered, and dispatch it."""
-        message = event.message
+        # don't waste time processing this if there are no prefixed commands
+        if not self.prefixed_commands:
+            return
+
+        data = event.data
 
-        if not message.content:
+        # many bots will not have the message content intent, and so will not have content
+        # for most messages. since there's nothing for prefixed commands to work off of,
+        # we might as well not waste time
+        if not data.get("content"):
             return
 
-        if not message.author.bot:
-            prefixes: str | Iterable[str] = await self.generate_prefixes(self, message)
+        # webhooks and users labeled with the bot property are bots, and should be ignored
+        if data.get("webhook_id") or data["author"].get("bot", False):
+            return
 
-            if isinstance(prefixes, str) or prefixes == MENTION_PREFIX:
-                # its easier to treat everything as if it may be an iterable
-                # rather than building a special case for this
-                prefixes = (prefixes,)  # type: ignore
+        # now, we've done the basic filtering out, but everything from here on out relies
+        # on a proper message object, so we do the same thing as the message create
+        # processor
+        message = self.cache.place_message_data(event.data)
```

Comment:
This will result in the message deserializing twice. This effectively doubles the resource use of a bot that has prefixed commands

Reply:
Perf branch mitigates this *slightly* due to its lazy deserializer, but its rather bad practice to be doing the same work twice. I'd be curious to see if its reliable to assume the message is already in the cache, as processors will always run before listeners

Reply:
I'll take a look - admittedly this uncertainty was the reason why I didn't rely on the cache in the first place, but there may be enough of a delay to make it work.

If it really comes down to it, the idea of making a really dumb loop to halt it while it waits for the message to appear is an option.

Reply:
Instead of a polling loop, why not:
- check if the message is already in the cache
- if not, wait for a message create event
- continue

Reply:
And done!